bundle/edu.gemini.ags.client.impl/src/main/scala/edu/gemini/ags/client/impl/AgsHttpClient.scala

```diff
@@ -39,12 +47,18 @@ case class AgsHttpClient(host: String, port: Int) extends AgsClient {
   private def estimateNow(url: URL): AgsResult = {
     try {
       AgsHttpClient.Log.info(s"AGS Query to $url")
-      val conn = url.openConnection().asInstanceOf[HttpsURLConnection]
-      conn.setHostnameVerifier(hostnameVerifier)
-      conn.setSSLSocketFactory(GemSslSocketFactory.get)
-      conn.setReadTimeout(timeout)
-      Charset.set(conn)
-      Response(conn).result
+      Option(lRUCache.get(url.toString) : AgsResult) | {
+        val conn = url.openConnection().asInstanceOf[HttpsURLConnection]
+        conn.setHostnameVerifier(hostnameVerifier)
+        conn.setSSLSocketFactory(GemSslSocketFactory.get)
+        conn.setReadTimeout(timeout)
+        Charset.set(conn)
+
+        Response(conn).result match {
+          case s@AgsResult.Success(_) => lRUCache.put(url.toString, s)
```

Comment:
Ugh, you're right. Sorry about that.

Would it make sense to calculate the value, wrap the cache update in an `IO[Unit]` to clarify that it has a side effect, and then `unsafePerformIO` on it, and then return the value, or would that be overkill? Trying to get a better handle on when I should be using `IO`. So something like:

`val result = ...
result match {
    case s: AgsResult.Success => IO { lruCache.put(url.toString, s) }.unsafePerformIO
    case _ =>
}
result`
?